### PR TITLE
refactor(auth): Improve auth pages structure

### PR DESCRIPTION
**Pull request by mkidawa**
closes: #43

### CODE REVIEW

File: src/pages/AuthPage/RegisterPage/components/RegisterForm/RegisterForm.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import { AuthFormWrapper } from 'pages/AuthPage/components/AuthFormWrapper/AuthFormWrappper';
+import { Button } from 'components/Button/Button';
+import { FormInput } from 'components/Form/FormInput/FormInput';
+import { Stack } from '@mui/material';
+import { useTranslation } from 'react-i18next';
+import { zodResolver } from '@hookform/resolvers/zod';
+
+type RegisterFormProps = {
+  onFormSubmit: SubmitHandler<RegisterFormData>;
+};
```
Comment (kaczor6418) [inline src/pages/AuthPage/RegisterPage/components/RegisterForm/RegisterForm.tsx:13]: can we move `RegisterFormProps` to *RegisterForm.types.ts* file?

File: src/pages/AuthPage/LoginPage/components/SignInProvidersBox/SignInProvidersBox.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
+import { AuthProvider } from 'services/FirebaseService/AuthService/AuthService.types';
+import { Box } from '@mui/material';
+import { Button } from 'components/Button/Button';
+import { useTranslation } from 'react-i18next';
+
+type SignInProvidersBoxProps = {
+  handleProviderButtonClick: (provider: AuthProvider) => void;
+};
```
Comment (kaczor6418) [inline src/pages/AuthPage/LoginPage/components/SignInProvidersBox/SignInProvidersBox.tsx:8]: can we move `SignInProvidersBoxProps` to *SignInProvidersBox.types.ts* file?

File: src/pages/AuthPage/LoginPage/components/LoginForm/LoginForm.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+import { Button } from 'components/Button/Button';
+import { FormInput } from 'components/Form/FormInput/FormInput';
+import { SignInProvidersBox } from '../SignInProvidersBox/SignInProvidersBox';
+import { useTranslation } from 'react-i18next';
+import { zodResolver } from '@hookform/resolvers/zod';
+
+type LoginFormProps = {
+  onFormSubmit: SubmitHandler<LoginFormData>;
+  handleExternalProviderLogin: (provider: AuthProvider) => void;
+};
```
Comment (kaczor6418) [inline src/pages/AuthPage/LoginPage/components/LoginForm/LoginForm.tsx:16]: can we move `LoginFormProps` to *LoginForm.types.ts* file?

**Review by kaczor6418 — CHANGES_REQUESTED** [review]

File: src/pages/AuthPage/RegisterPage/components/RegisterForm/RegisterForm.tsx (hunk 1 above)
Comment (kaczor6418) [reply src/pages/AuthPage/RegisterPage/components/RegisterForm/RegisterForm.tsx:13]: just to follow structure described in docmentation:
https://github.com/GDSC-Lodz-University-of-Technology/warzywniax/wiki/Project-structure

File: src/pages/AuthPage/LoginPage/components/SignInProvidersBox/SignInProvidersBox.tsx (hunk 2 above)
Comment (kaczor6418) [reply src/pages/AuthPage/LoginPage/components/SignInProvidersBox/SignInProvidersBox.tsx:8]: just to follow structure described in docmentation:
https://github.com/GDSC-Lodz-University-of-Technology/warzywniax/wiki/Project-structure

File: src/pages/AuthPage/LoginPage/components/LoginForm/LoginForm.tsx (hunk 3 above)
Comment (kaczor6418) [reply src/pages/AuthPage/LoginPage/components/LoginForm/LoginForm.tsx:16]: just to follow the structure described in documentation:
https://github.com/GDSC-Lodz-University-of-Technology/warzywniax/wiki/Project-structure

**Review by kaczor6418 — APPROVED** [review]
LGTM :sunglasses:

**Review by HKGx — APPROVED** [review]
Looking good :+1:

**Pull request merged**